Scripts/PAIF-N/01-deploy-vcf-workload-domain.ps1

```diff
@@ -0,0 +1,345 @@
+The commissioning/creation specs are constructed. The specs are validated. The actual operation is invoked.
+The validation and operation invocation are long-running tasks. This requires awaiting and status tracking
+until their completion. The waiting for validation and the actual operation is done using helper cmdlets -
+Wait-VcfValidation and Wait-VcfTask, located in utils sub-folder.
+
+On completion a new VI workload domain reflecting the given parameters should be created.
+
+.NOTES
+
+Prerequisites:
+ - A VCF Management Domain
+ - A minimum of three free hosts marked with the appropriate storage
+
+"Global parameters", "Host commissioning parameters", "Workload domain creation parameters" should be updated to
+reflect the environment they are run in. This may require altering the spec creation script.
+
+#>
+
+$ErrorActionPreference = 'Stop'
+$SCRIPTROOT = ($PWD.ProviderPath, $PSScriptRoot)[!!$PSScriptRoot]
+. (Join-Path $SCRIPTROOT 'utils/Wait-VcfTask.ps1')
+. (Join-Path $SCRIPTROOT 'utils/Wait-VcfValidation.ps1')
+
+
+# Configure PowerCLI
+Set-PowerCLIConfiguration -Scope User -ParticipateInCEIP $false -Confirm:$false
+Set-PowerCLIConfiguration -scope user -InvalidCertificateAction Ignore -Confirm:$false
+
+# --------------------------------------------------------------------------------------------------------------------------
+# Global parameters
+# --------------------------------------------------------------------------------------------------------------------------
+
+$OrgName = 'VMware'
+$domainName = 'sfo-w01'
+
+$domain = 'vrack.vsphere.local'
+$gateway = '10.0.0.250'
+$sddcManager = @{
+   Fqdn = "sddc-manager.$domain"
+   User = 'administrator@vsphere.local'
+   Password = 'VMware123!'
+}
+
+############################################################################################################################
+# Host commissioning
+############################################################################################################################
+
+# --------------------------------------------------------------------------------------------------------------------------
+# Host commissioning parameters
+
+$esxiHosts = @(
+   @{
+      Fqdn = "esxi-5.$domain"
+      Username = 'root'
+      Password = "ESXiSddc123!"
+      StorageType = "VSAN"
+      LicenseKey = 'XXXXX-XXXXX-XXXXX-XXXXX-XXXXX'
+   }
+   @{
+      Fqdn = "esxi-6.$domain"
+      Username = 'root'
+      Password = "ESXiSddc123!"
+      StorageType = "VSAN"
+      LicenseKey = 'XXXXX-XXXXX-XXXXX-XXXXX-XXXXX'
+   }
+   @{
+      Fqdn = "esxi-7.$domain"
+      Username = 'root'
+      Password = "ESXiSddc123!"
+      StorageType = "VSAN"
+      LicenseKey = 'XXXXX-XXXXX-XXXXX-XXXXX-XXXXX'
+   }
+   @{
+      Fqdn = "esxi-8.$domain"
+      Username = 'root'
+      Password = "ESXiSddc123!"
+      StorageType = "VSAN"
+      LicenseKey = 'XXXXX-XXXXX-XXXXX-XXXXX-XXXXX'
+   }
+)
+
+$networkPoolName = 'networkpool'
```

Comment:
Same as above, should this be documented?

Reply:
I've added the documentation from the API docs